### PR TITLE
Fix NullPointerException for Stats.getStatsContextFactory()

### PR DESCRIPTION
**Pull request by songy23**
Please note Provider.newInstance() is Nullable.

### CODE REVIEW

File: core/src/main/java/com/google/instrumentation/stats/Stats.java [hunk 1]
```diff
@@ -26,7 +26,7 @@
    * Returns the default {@link StatsContextFactory}.
    */
   public static StatsContextFactory getStatsContextFactory() {
-    return statsManager.getStatsContextFactory();
+    return statsManager == null ? null : statsManager.getStatsContextFactory();
```
Comment (bogdandrutu) [inline core/src/main/java/com/google/instrumentation/stats/Stats.java]: Shouldn't we return a no-op factory in this case? That is what we do in trace.

Comment (songy23) [inline core/src/main/java/com/google/instrumentation/stats/Stats.java]: Create a separate issue for this: https://github.com/google/instrumentation-java/issues/215

Comment (bogdandrutu) [inline core/src/main/java/com/google/instrumentation/stats/Stats.java]: Please use at least the "@Nullable" annotation.

Comment (sebright) [inline core/src/main/java/com/google/instrumentation/stats/Stats.java]: There is already an issue requesting a no-op implementation: #41

/cc @zhangkun83

Comment (songy23) [inline core/src/main/java/com/google/instrumentation/stats/Stats.java]: Fixed.

Comment (zhangkun83) [inline core/src/main/java/com/google/instrumentation/stats/Stats.java]: @sebright We have made our code to skip reporting logic if `getStatsContextFactory()` returns null, so we don't need the no-op impl any more. I am going to close that issue.

**Pull request merged**